source/ibl_sampler.js

```diff
@@ -105,6 +160,26 @@ class iblSampler
             return texture;
         }
 
+        if(this.supportedFormat == "HALF_FLOAT")
+        {
+            texture.internalFormat = this.internalFormat();
+            texture.format = this.gl.RGBA;
+            texture.type = this.type();
+
+            const numPixels = image.dataFloat.length / 3;
+            texture.data = new Uint16Array(numPixels * 4);
```

Comment:
It is valid to upload 32-bit float data to 16-bit float textures. The conversion will be done automatically.

For example, a `RGBA16F` texture can be created with:
```js
gl.texImage2D(gl.TEXTURE_2D, 0, gl.RGBA16F, width, height, 0, gl.RGBA, gl.FLOAT, myFloat32array);
```